envconfig.go

```diff
@@ -104,6 +105,17 @@ func (h FieldTypeHandler) parserNames() []string {
 	return ret
 }
 
+// expand uses os.Expand and the given lookupFunc to expand ${xxx} constructs
+// in the given value.
+func expand(value string, lookupFunc func(string) (string, bool)) string {
+	return os.Expand(value, func(key string) string {
+		if v, ok := lookupFunc(key); ok {
+			return v
+		}
+		return ""
+	})
+}
```

Comment:
Having the default here be an empty string... It's not bad, but I prickle at it. Would it be possible to do similar as `defaultFrom` to have it reference a different struct field?

Reply:
I don't see the recursion. `ev` origins from `lookup` which is the original environment lookup. It's expanded once on line 259, and then passed on to the parser. The resulting `val` is never expanded.

Reply:
I don't want environment expansion to reference a struct field. That would IMO violate what environment expansion is all about. It expands environment variables using the scope of the environment. The `defaultFrom` is different, because it fetches an already parsed value and uses that as the default without parsing it again, which is fine. Different scope (on the other side of the parsers).